lib/Service/GroupsIntegration.php

```diff
@@ -35,8 +37,11 @@ class GroupsIntegration {
 	 */
 	private $groupServices = [];
 
-	public function __construct(IGroupService $groupService) {
-		$this->groupServices[] = $groupService;
+	public function __construct(ContactsGroupService $contactsGroupService, NextcloudGroupService $nextcloudGroupService) {
+		$this->groupServices = [
```

Comment:
So we decide on ordering of the services here?

If any service fails nothing is returned in getMatchingGroups, but that is outside the scope of this issue.

Reply:
Code looks good.

However, I wanted to do an integration test on the GUI, but I'm not able to install the mail app since `composer install` fails on requiring `kwi/urllinker` which seems to be a private repo?

Reply:
> fails on requiring `kwi/urllinker` which seems to be a private repo?

Are you sure it fails? I think that is just a warning. You can perhaps ignore that.

Reply:
> If any service fails nothing is returned in getMatchingGroups, but that is outside the scope of this issue.

Yes and yes :)